### PR TITLE
Update init.sh to use go-init

### PR DESCRIPTION
**Pull request by gregakinman**
fysa @cbrockington

### CODE REVIEW

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.groovy [hunk 1]
```diff
@@ -46,7 +46,8 @@ class JavaServiceDistributionPlugin implements Plugin<Project> {
 
         project.configurations.create('goJavaLauncherBinaries')
         project.dependencies {
-            goJavaLauncherBinaries 'com.palantir.launching:go-java-launcher:1.2.0'
+            // TODO (gregakinman): update to 1.4.0 once released
```
Comment (gregakinman) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.groovy:49]: blocked on merge and release of https://github.com/palantir/go-java-launcher/pull/49

File: gradle-sls-packaging/src/main/resources/init.sh [hunk 2]
```diff
@@ -52,78 +54,20 @@ STATIC_LAUNCHER_CONFIG="service/bin/launcher-static.yml"
 CUSTOM_LAUNCHER_CONFIG="var/conf/launcher-custom.yml"
 STATIC_LAUNCHER_CHECK_CONFIG="service/bin/launcher-check.yml"
 
+DEPRECATION_MESSAGE="Command is deprecated: the next major release of sls-packaging will only support start/status/stop"
```
Comment (gregakinman) [inline gradle-sls-packaging/src/main/resources/init.sh:43]: Don't have to do this now & am not married to it, but thought I'd throw it in

File: gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy [hunk 3]
```diff
@@ -176,6 +150,7 @@ class ServiceDistributionPluginTests extends GradleTestSpec {
 
         then:
         execAllowFail('dist/service-name-0.0.1/service/bin/init.sh', 'start')
+        sleep 1000
```
Comment (gregakinman) [inline gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy:155]: for some reason this was necessary to make this pass locally?? ¯\_(ツ)_/¯

**Comment by gregakinman** [comment]
@dansanduleac can I get write perms on this repo so I can debug the Circle builds with SSH? They're working on my machine but I need to debug why they're not working on Circle.

**Comment by gregakinman** [comment]
nvm, I realized it was because I was being dumb and in my local iteration had only been copying over the darwin binary. have now checked in the updated linux binary. (of course this will be a non-issue once https://github.com/palantir/go-java-launcher/pull/49 merges).

File: gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/GradleTestSpec.groovy [hunk 4]
```diff
@@ -67,20 +67,24 @@ class GradleTestSpec extends Specification {
...
+        int expected = 0
         Assert.assertEquals(sprintf("Expected command '%s' to exit with '%d'", tasks.join(' '), expected), expected, result)
         sleep 1000 // wait for the Java process to actually run
         return sout.toString()
     }
 
+    protected int execWithExitCode(String... tasks) {
+        Process proc = new ProcessBuilder().command(tasks).directory(projectDir).start()
+        int result = proc.waitFor()
+        sleep 1000
```
Comment (dansanduleac) [inline gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/GradleTestSpec.groovy]: why is this necessary?

File: gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy (hunk 3 above)
Comment (dansanduleac) [reply gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy:155]: shouldn't the process finish by the time the above function returns?

Comment (dansanduleac) [reply gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy:155]: ah nvm - this makes sense here because I guess `start` exits right away, whereas the init.sh script was doing `sleep 5` after starting the process

File: gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy [hunk 5]
```diff
@@ -63,15 +63,14 @@ class ServiceDistributionPluginTests extends GradleTestSpec {
...
-        exec('dist/service-name-0.0.1/service/bin/init.sh', 'start') ==~ /(?m)Running 'service-name'\.\.\.\s+Started \(\d+\)\n/
-        file('dist/service-name-0.0.1/var/log/service-name-startup.log', projectDir).text.contains('Test started\n')
-        exec('dist/service-name-0.0.1/service/bin/init.sh', 'start') ==~ /(?m)Process is already running\n/
-        exec('dist/service-name-0.0.1/service/bin/init.sh', 'status') ==~ /(?m)Checking 'service-name'\.\.\.\s+Running \(\d+\)\n/
-        exec('dist/service-name-0.0.1/service/bin/init.sh', 'restart') ==~
-                /(?m)Stopping 'service-name'\.\.\.\s+Stopped \(\d+\)\nRunning 'service-name'\.\.\.\s+Started \(\d+\)\n/
-        exec('dist/service-name-0.0.1/service/bin/init.sh', 'stop') ==~ /(?m)Stopping 'service-name'\.\.\.\s+Stopped \(\d+\)\n/
-        exec('dist/service-name-0.0.1/service/bin/init.sh', 'check') ==~ /(?m)Checking health of 'service-name'\.\.\.\s+Healthy.*\n/
-        exec('dist/service-name-0.0.1/service/monitoring/bin/check.sh') ==~ /(?m)Checking health of 'service-name'\.\.\.\s+Healthy.*\n/
+        execWithExitCode('dist/service-name-0.0.1/service/bin/init.sh', 'start') == 0
```
Comment (dansanduleac) [inline gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy:66]: Think we should just add a `sleep 1000` after this line (to wait for the process to be up), and remove the sleeps in the util methods - execWithExitCode and execWithOutput

File: gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy (hunk 3 above)
Comment (gregakinman) [reply gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/service/ServiceDistributionPluginTests.groovy:155]: ahhh that's why

**Review by dansanduleac — APPROVED** [review]

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.groovy (hunk 1 above)
Comment (gregakinman) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.groovy:49]: tracking at https://github.com/palantir/sls-packaging/issues/308

**Comment by iamdanfox** [comment]
@gregakinman @dansanduleac hey guys I don't think it's really acceptable to check in the `gradle-sls-packaging/src/main/resources/linux-amd64/go-init` binary.

Was that binary built from develop of palantir/go-java-launcher (containing functionality from [this merged PR](https://github.com/palantir/go-java-launcher/pull/45))? If so, please could you just tag a release of go-java-launcher and delete the checked in binary ASAP (it seems there are [40 unmerged commits on develop](https://github.com/palantir/go-java-launcher/compare/1.3.0...develop))?

Or does the binary contain functionality from the unmerged https://github.com/palantir/go-java-launcher/pull/49 PR?  In this case, I think we need to do a short-term revert here. 

Infra repos affect a huge number of projects here, so we need to maintain the highest possible bar for good-practises / general repo hygiene. The particular situation I'm concerned about is that we wouldn't be able to fix an urgent bug with a small PR onto develop of palantir/go-java-launcher.

**Comment by gregakinman** [comment]
Makes sense. I'm waiting for @dansanduleac to +1 https://github.com/palantir/go-java-launcher/pull/49 in which case we can merge it and fix https://github.com/palantir/sls-packaging/issues/308 to fix this. Or, depending on how long it will take DanS, if he thinks it will take him longer than 1 day, then we can revert this now.
